src/soc-support.md

```diff
@@ -0,0 +1,183 @@
+
+### Flash Algorithms
+
+[Flash Algorithms](https://open-cmsis-pack.github.io/Open-CMSIS-Pack-Spec/main/html/flashAlgorithm.html)
+are integrated with debugging tools like `probe-rs`. They facilitate and speed
+up firmware programming and debugging, streamlining development workflows.
+Providing well-supported FlashAlgos will enhance the integration with these
+tools and improve the overall developer experience.
+
+### Vendor Tooling
+
+Some System-on-Chip (SoC) devices require custom tools for generating images or
+flashing them onto the device. It is beneficial to provide these tools in an
+open-source manner, fostering community contributions and accelerating
+ecosystem growth. Open-sourcing vendor tooling enables third-party developers
+to extend and enhance the toolchain, ensuring improved compatibility with the
+broader Embedded Rust ecosystem.
+
+### Contact Information
+
+Providing contact information is vital for addressing maintainer queries and
+issues related to register description files or other resources. The use of a
+public issue tracking system (like GitHub Issues) for reporting and tracking
+problems might help. Actively engage with the community through forums,
+discussions, and updates to build trust and collaboration.
+
+## Maintaining PAC and HAL Crates
+
+Peripheral Access Crates (PACs) and Hardware Abstraction Layer (HAL) crates are
+at the core of enabling Rust support.
+
+### Generate and Maintain PACs
+
+Multiple tools such as `svd2rust`, `chiptool`, `raltool`, and `svd2pac`
+automate the generation of PACs from register description files. Each tool has
+its strengths, and selecting the right one depends on the requirements and the
+complexity of the hardware.
+
+### Develop and Maintain HAL Crates
+
+Implement `embedded-hal` and `embedded-hal-async` traits in your HAL crates.
+Adhering to these traits ensures compatibility across the Embedded Rust
+ecosystem, enhancing interoperability. It is an essential goal that HALs use
+Rust code rather than wrapping existing C code. An incremental porting
+strategy, where Rust is used for all core functionality, but C with Rust
+bindings is used for complex drivers, is acceptable, allowing for gradual
+adoption and community contributions.
+
+Start with essential peripherals (clock, timer, GPIO) and expand progressively
+(I2C, SPI, UART, etc.) based on community feedback. Release early and often to
+engage the community and gather valuable insights for further development.
+
+### Common Recommendations
+
+- Ensure that crates are compatible with `no_std` environments, which are
+  common in embedded systems without an operating system. Functionality that
+  needs `alloc` or `std` can be included when gated with Cargo "features."
+- Make your crates available on [crates.io](https://crates.io/) to maximize
+  visibility and ease of use for developers.
+- Use semantic versioning to maintain consistency and predictability in your
```

Comment:
general feedback: the more links the better! (if they point to stable URLs, anyway)
don't expect the reader to know these terms (they might be managers or come from an otherwise vastly different background) and might not even realise that this is meaning something specific and that they could/should look it up
```suggestion
- Use [semantic versioning](https://semver.org/) to maintain consistency and predictability in your
```

Reply:
I do worry that the page will end up having 10% of the words being links 😂